src/zm_ffmpeg_camera.cpp

```diff
@@ -870,14 +872,23 @@ int FfmpegCamera::CaptureAndRecord(
           // since the last keyframe.
           unsigned int packet_count = 0;
           ZMPacket *queued_packet;
+          struct timeval video_offset = {0};
 
           // Clear all packets that predate the moment when the recording began
           packetqueue->clear_unwanted_packets(
-              &recording, monitor->GetPreEventCount(), mVideoStreamId);
+              &recording, 0, mVideoStreamId);
```

Comment:
Wait, won't this make it clear too many packets?

Reply:
The original would have been correct if the start-time passed as `recording` did not include the pre-event count. But it does already include it (see below). So we should not keep an additional pre-event-count frames before the start-time (=`recording`).

Note: the value of `recording` passed to` CaptureAndRecording()` is
`videostore_data->recording = event->StartTime()`. Event is created with 
`start-time = *(image_buffer[pre_index].timestamp)`
where pre-index included pre-event frame count.

An exception is when a new recording segment is created due to segment length exceeding the limit. But in that case no pre-event count back-track required.

Reply:
right, I see now. Cool!